### PR TITLE
Add types for hbs npm package

### PR DESCRIPTION
**Pull request by DavidM77**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Notes: Function type is used, but this is inline with the existing Handlebars and Express types.  Since hbs is an integration between Handlebars and Express I feel this is justified.

### CODE REVIEW

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Comment by DavidM77** [comment]
The test contains the following idiomatic express:
app.engine('html', require('hbs').__express);

If I use the suggested tslint.json, the tests fail with the following:
ERROR: 6:20  no-var-requires  require statement not part of an import statement

I not sure how to resolve this issue without adding to tslint.json.

**Comment by plantain-00** [comment]
You can add `// tslint:disable-next-line no-var-requires` above the line of code

**Review by plantain-00 — APPROVED** [review]

File: types/hbs/hbs-tests.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import * as express from 'express';
+
+const app = express();
+
+app.set('view engine', 'html');
+// tslint:disable-next-line no-var-requires
+app.engine('html', require('hbs').__express);
```
Comment (andy-ms) [inline types/hbs/hbs-tests.ts]: Actually, this lint rule is important -- `require()` is a function returning `any`. Please use an `import hbs = require("hbs")` instead.

**Review by ghost — CHANGES_REQUESTED** [review]
See comment

**Pull request merged**